src/app/lab/components/model-modal.component.ts

```diff
@@ -46,10 +58,15 @@ export class ModelModalComponent implements OnInit {
     }
 
     public updateProjectClimateModels() {
+        this.disableClimateModels();
         this.projectData.models = this.filterSelectedClimateModels();
         this.updateButtonText();
```

Comment:
updateButtonText() is broken because the check is for # models selected against the full list of models.

this.projectData.models.length === this.climateModels.filter(m=> m.datasets.includes(this.projectData.dataset.name)).length

Reply:
Good find. Can just use the count of project `dataset` models instead.